astrokat/observe_main.py

```diff
@@ -534,7 +545,10 @@ def run_observation(opts, kat):
                         "TRACE: initial observer for target\n {}".format(observer)
                     )
                     # check target visible before doing anything
-                    if not above_horizon(katpt_target, horizon=opts.horizon):
+                    # make sure the target would be visible for the entire duration
+                    target_duration = target['duration']
```

Comment:
This duration excludes the time taken to observe the cadence targets, which happens before `target` is observed.  If it takes "long" for the cadence targets to be observed, then `target` may set during its actual observation, even though the check now says it will be up.

Is the time to observe cadence targets significant, i.e., "long"?  I guess that it would vary widely, depending what the astronomer had in mind.

We could add another check for `above_horizon` after the cadence targets loop to be more accurate.  This would only help for astrokat simulation (see next paragraph).

In CAM executions using session the `track`, `scan` and `raster_scan` methods include a check for targets above the horizon, while the astrokat simulation implementation of those method does not.  This will still give slightly different results.

Reply:
It is correct that it does not include cadence observation time. But cadence is no longer supported by OPT, and the fix to include cadence is complicated.